command-line/lib/Command/Key/WalletId.hs

```diff
@@ -0,0 +1,79 @@
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE TypeSynonymInstances #-}
+
+{-# OPTIONS_HADDOCK hide #-}
+
+module Command.Key.WalletId
+    ( Cmd (..)
+    , mod
+    , run
+    ) where
+
+import Prelude hiding
+    ( mod )
+
+import Cardano.Address.Derivation
+    ( hashCredential, toXPub, xprvFromBytes, xpubToBytes )
+import Codec.Binary.Encoding
+    ( AbstractEncoding (..) )
+import Control.Monad
+    ( when )
+import Options.Applicative
+    ( CommandFields, Mod, command, footerDoc, helper, info, progDesc )
+import Options.Applicative.Help.Pretty
+    ( string )
+import System.IO
+    ( stdin, stdout )
+import System.IO.Extra
+    ( hGetBech32, hPutBytes )
+
+import qualified Cardano.Codec.Bech32.Prefixes as CIP5
+import qualified Data.ByteString as BS
+
+data Cmd = WalletId
+    deriving (Show)
+
+mod :: (Cmd -> parent) -> Mod CommandFields parent
+mod liftCmd = command "walletid" $
+    info (helper <*> fmap liftCmd parser) $ mempty
+        <> progDesc "Get the wallet id from a given key"
+        <> footerDoc (Just $ string $ mconcat
+            [ "The key is read from stdin."
+            ])
+  where
+    parser = pure WalletId
+
+run :: Cmd -> IO ()
+run WalletId = do
+    (hrp, bytes) <- hGetBech32 stdin allowedPrefixes
```

Comment:
Perhaps in a future version we could accept hex-encoded vks?
I know this would not be able to validate the type of key, but it might be helpful for users to have a back door.